src/main/java/nl/novi/TechItEasy/controllers/TelevisionController.java

```diff
@@ -0,0 +1,116 @@
+package nl.novi.TechItEasy.controllers;
+
+import nl.novi.TechItEasy.Repositories.TelevisionRepository;
+import nl.novi.TechItEasy.exceptions.RecordNotFoundException;
+import nl.novi.TechItEasy.models.Television;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/televisions")
+public class TelevisionController {
+    @Autowired
+    TelevisionRepository televisionRepository;
+
+    @GetMapping
+    public ResponseEntity<Object> getAllTelevisions() {
+        if (televisionRepository.count() > 0) {
+            return new ResponseEntity<>(televisionRepository.findAll(), HttpStatus.OK);
+        } else {
+            throw new RecordNotFoundException("No televisions found");
+        }
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Television> getTelevision(@PathVariable int id) {
+        if (televisionRepository.existsById(id)) {
+            Optional <Television> optionalTelevision = televisionRepository.findById(id);
+            return new ResponseEntity<>(optionalTelevision.get(), HttpStatus.OK);
+        } else {
+            throw new RecordNotFoundException(id + " not found");
+        }
+    }
+
+    @PostMapping
+    public ResponseEntity<Object> createTelevision(@RequestBody Television television) {
+            if (televisionRepository.existsByName(television.getName())) {
+                throw new RecordNotFoundException("Television is allready in our list!");
+            }
+        televisionRepository.save(television);
+        return new ResponseEntity<>(television, HttpStatus.CREATED);
+    }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<Object> updateTelevision(@PathVariable int id, @RequestBody Television television) {
+        Optional<Television> updatedTelevision = televisionRepository.findById(id);
+        if (updatedTelevision.isPresent()) {
+            Television newTelevision = updatedTelevision.get();
+            if (television.getType() != null) {
```

Comment:
In de uitwerking wordt niet op aanwezigheid van een waarde gecheckt. Er valt wat voor en tegen te zeggen. 
Het kan zijn dat iemand bewust het type leeg aanlevert, en daarmee het typeveld wil leegmaken (zelfde geldt voor andere velden).  Dit is in jouw variant niet mogelijk.